src/Files.App/Utils/Shell/Win32API.cs

```diff
@@ -835,6 +835,27 @@ public static Task InstallFont(string fontFilePath, bool forAllUsers)
 			return RunPowershellCommandAsync($"-command \"Copy-Item '{fontFilePath}' '{fontDirectory}'; New-ItemProperty -Name '{Path.GetFileNameWithoutExtension(fontFilePath)}' -Path '{registryKey}' -PropertyType string -Value '{destinationPath}'\"", forAllUsers);
 		}
 
+		public static Task InstallFonts(string[] fontFilePaths, bool forAllUsers)
+		{
+			string fontDirectory = forAllUsers
+				? Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.Windows), "Fonts")
+				: Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData), "Microsoft", "Windows", "Fonts");
+
+			string registryKey = forAllUsers
+				? "HKLM:\\Software\\Microsoft\\Windows NT\\CurrentVersion\\Fonts"
+				: "HKCU:\\Software\\Microsoft\\Windows NT\\CurrentVersion\\Fonts";
+
+			string psCommand = string.Empty;
+
+			foreach (string fontFilePath in fontFilePaths)
+			{
+				var destinationPath = Path.Combine(fontDirectory, Path.GetFileName(fontFilePath));
+				psCommand = psCommand + $"-command \"Copy-Item '{fontFilePath}' '{fontDirectory}'; New-ItemProperty -Name '{Path.GetFileNameWithoutExtension(fontFilePath)}' -Path '{registryKey}' -PropertyType string -Value '{destinationPath}'\";";
+			}
+
+			return RunPowershellCommandAsync(psCommand, forAllUsers);
+		}
```

Comment:
@hishitetsu I just noticed the suggested change moved `RunPowershellCommandAsync(psCommand.Append("\"").ToString(), forAllUsers);` back into the loop causing #14503, is there a specific reason to do it like this?

Reply:
`RunPowershellCommandAsync` is called in the loop only when the command becomes too long. Thus, the command should normally be called only once.
It may be a good idea to always invoke PowerShell with administrator privileges when installing fonts.